File: $_algotsu-kaizen/problemo-set/count-the-number-of-consistent-strings.c

int countConsistentStrings(char * allowed, char ** words, int wordsSize) {
    int idx = 0b0;
    unsigned int mask = 0b0, bit = 0b1;
    while (*(allowed + idx)) mask |= bit << (*(allowed + idx++) - 'a');

    int count = 0b0;
    for (int i = 0b0; i < wordsSize; ++i) {
        bool flag = true;
        idx = 0b0;
        while ( *(*(words + i) + idx) ) {
            if ((bit << (*(*(words + i) + idx++) - 'a')) & mask) {
                continue;
            } else {
                flag = false;
                break;
            }
        }
        flag && ++count;
    }

    return count;
}

File: $_algotsu-kaizen/problemo-set/count-the-number-of-consistent-strings.rs

impl Solution {
    pub fn count_consistent_strings(allowed: String, words: Vec<String>) -> i32 {
        let mut count: i32 = 0b0;
        let mut chars = std::collections::HashSet::new();

        for c in allowed.chars() { 
            chars.insert(c);
        }

        for word in words { 
            let mut flag = true;
            for c in word.chars() { 
                if chars.contains(&c) {
                    continue;
                }

                flag = false;
                break;
            }

            if flag {
                count += 1;
            }
        }

        return count;
    }
}
